protocols/gossipsub/src/behaviour.rs

```diff
@@ -1885,20 +1848,17 @@ where
                 metrics.register_invalid_message(&raw_message.topic);
             }
 
-            let fast_message_id_cache = &self.fast_message_id_cache;
+            if let Ok(message) = self.data_transform.inbound_transform(raw_message.clone()) {
+                let message_id = self.config.message_id(&message);
```

Comment:
It's already done before calling `handle_invalid_message`.
Need to check how to handle peer_score

Reply:
Alright, imo after checking the behaviour this should stay as it is.

Reply:
Great! Thanks for looking into it :)